src/crypto/rand/rand_esp32c3.go

```diff
@@ -0,0 +1,59 @@
+// +build esp32c3
+
+// This implementation of crypto/rand uses on-chip random generator
+// to generate random numbers.
+//
+
+package rand
+
+import (
+	"device/esp"
+	"unsafe"
+)
+
+func init() {
+	Reader = &reader{}
+
+	// When using the random number generator, make sure at least either the SAR ADC,
+	// high-speed ADC1, or RTC20M_CLK2 is enabled. Otherwise, pseudo-random numbers will be returned.
+	//  • SAR ADC can be enabled by using the DIG ADC controller. For details,
+	//    please refer to Chapter 6 On-Chip Sensors and Analog Signal Processing [to be added later].
+	//  • High-speed ADC is enabled automatically when the Wi-Fi or Bluetooth modules
+	//    is enabled.
+	//  • RTC20M_CLK is enabled by setting the RTC_CNTL_DIG_CLK20M_EN bit in
+	//    the RTC_CNTL_CLK_CONF_REG register.
+	// Note:
+	//  1. Note that, when the Wi-Fi module is enabled, the value read from the high-speed
+	//     ADC can be saturated in some extreme cases, which lowers the entropy. Thus, it
+	//     is advisable to also enable the SAR ADC as the noise source for the random
+	//     number generator for such cases.
+	//  2. Enabling RTC20M_CLK increases the RNG entropy. However, to ensure maximum entropy,
+	//     it’s recommended to always enable an ADC source as well.
+
+	// Enable SAR ADC
+	esp.SYSTEM.PERIP_CLK_EN0.SetBits(esp.SYSTEM_PERIP_CLK_EN0_APB_SARADC_CLK_EN)
```

Comment:
It appears that enabling random number generation is a bit more complicated than this:

https://github.com/espressif/esp-idf/blob/v4.3/components/bootloader_support/src/bootloader_random_esp32c3.c

I'm worried that it is not truly random without this complete initialization.

Reply:
I will move it to draft until have move details on initialization and the  #2325 with #2333 are move so I would have better access to register's fields.